pysonos/events.py

```diff
@@ -186,11 +188,32 @@ def listen(self, ip_address):
             The port on which the event listener listens is configurable.
             See `config.EVENT_LISTENER_PORT`
         """
-        address = (ip_address, self.requested_port_number)
-        self._listener_thread = EventServerThread(address)
+        for port_number in range(self.requested_port_number,
+                                 self.requested_port_number + 100):
+            address = (ip_address, port_number)
+            try:
+                server = EventServer(address, EventNotifyHandler)
+                break
+            except OSError as oserror:
+                if oserror.errno == errno.EADDRINUSE:
+                    log.warning(
+                        "Port %s:%d is in use",
+                        ip_address,
+                        port_number
+                    )
+                else:
+                    raise
+
+        self._listener_thread = EventServerThread(server)
         self._listener_thread.daemon = True
         self._listener_thread.start()
-        return self.requested_port_number
+        if port_number != self.requested_port_number:
+            log.warning(
```

Comment:
I changed all logs to `debug` to not have them pollute the Home Assistant log files by default and I wonder whether this message will make sense to most Home Assistant users.

With this fix, things will no longer break, so perhaps a diagnostic is not needed by default? 🤔

Reply:
Changed to debug